pkg/client/restclient/plugin.go

```diff
@@ -30,9 +30,16 @@ type AuthProvider interface {
 	// WrapTransport allows the plugin to create a modified RoundTripper that
 	// attaches authorization headers (or other info) to requests.
 	WrapTransport(http.RoundTripper) http.RoundTripper
+	// Login is expected to maybe require user interaction; it won't be called
+	// in environments where a user is not present
+	Login() error
```

Comment:
I'm expecting that it *might* require user interaction, just not through the command line (passing an io.Reader). I'll change the comment to try to reflect this. I'm mostly going to defer its usage (and any necessary changes) to @bobbyrullo, because it isn't currently used anywhere.